processor/schemaprocessor/config.go

```diff
@@ -0,0 +1,74 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schemaprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor"
+
+import (
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/config"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/schema"
+)
+
+var (
+	errRequiresTargets  = errors.New("requires schema targets")
+	errDuplicateTargets = errors.New("duplicate targets detected")
+)
+
+// Config defines the user provided values for the Schema Processor
+type Config struct {
+	config.ProcessorSettings `mapstructure:",squash"`
+
+	// PreCache is a list of schema URLs that are downloaded
+	// and cached at the start of the collector runtime
+	// in order to avoid fetching data that later on could
+	// block processing of signals. (Optional field)
+	PreCache []string `mapstructure:"precache"`
+
+	// Targets define what schema families should be
+	// translated to, allowing older and newer formats
+	// to conform to the target schema indentifier.
+	Targets []string `mapstructure:"targets"`
```

Comment:
What happens when I have two targets with conflicting conventions? Is the first taking precedence? The last? The latest following semvem?

Reply:
An error is returned during the validation step if there is duplicate schema families to avoid this.

Reply:
Could you document this as part of the readme, possibly with examples?